src/test/java/htsjdk/samtools/cram/build/Sam2CramRecordFactoryTest.java

```diff
@@ -0,0 +1,171 @@
+        Assert.assertEquals(refBases[0], substitution.getReferenceBase());
+
+        readFeatures.clear();
+        final Sam2CramRecordFactory s2mFactory_v3 = new Sam2CramRecordFactory(refBases, header, CramVersions.CRAM_v3);
+        s2mFactory_v3.addSubstitutionsAndMaskedBases(record, readFeatures, fromPosInRead, alignmentStartOffset, nofReadBases, readBases, scores);
+        Assert.assertEquals(1, readFeatures.size());
+
+        rf = readFeatures.get(0);
+        Assert.assertTrue(rf instanceof Substitution);
+        substitution = (Substitution) rf;
+        Assert.assertEquals(1, substitution.getPosition());
+        Assert.assertEquals(readBases[0], substitution.getBase());
+        Assert.assertEquals(refBases[0], substitution.getReferenceBase());
+    }
+
+    /**
+     * Test the outcome of non-ACGTN ref and read bases mismatching each other.
+     * The result should be explicit read base and score capture via {@link ReadBase}.
+     */
+    @Test
+    public void testAddSubstitutionsAndMaskedBasesAmbiguityMismatch() {
+        final byte[] refBases = "R".getBytes();
+        final byte[] readBases = "F".getBytes();
+        final byte[] scores = SAMUtils.fastqToPhred("!");
+
+        final SAMFileHeader header = new SAMFileHeader();
+        final CramCompressionRecord record = new CramCompressionRecord();
+        record.alignmentStart = 1;
+        final List<ReadFeature> readFeatures = new ArrayList<>();
+        final int fromPosInRead = 0;
+        final int alignmentStartOffset = 0;
+        final int nofReadBases = 1;
+
+        final Sam2CramRecordFactory s2mFactory_v2 = new Sam2CramRecordFactory(refBases, header, CramVersions.CRAM_v2_1);
+        s2mFactory_v2.addSubstitutionsAndMaskedBases(record, readFeatures, fromPosInRead, alignmentStartOffset, nofReadBases, readBases, scores);
+        Assert.assertEquals(1, readFeatures.size());
+
+        ReadFeature rf = readFeatures.get(0);
+        Assert.assertTrue(rf instanceof ReadBase);
+        ReadBase readBaseFeature = (ReadBase) rf;
+        Assert.assertEquals(1, readBaseFeature.getPosition());
+        Assert.assertEquals(readBases[0], readBaseFeature.getBase());
+        Assert.assertEquals(scores[0], readBaseFeature.getQualityScore());
+
+
+        readFeatures.clear();
+        final Sam2CramRecordFactory s2mFactory_v3 = new Sam2CramRecordFactory(refBases, header, CramVersions.CRAM_v3);
+        s2mFactory_v3.addSubstitutionsAndMaskedBases(record, readFeatures, fromPosInRead, alignmentStartOffset, nofReadBases, readBases, scores);
+        Assert.assertEquals(1, readFeatures.size());
+
+        rf = readFeatures.get(0);
+        Assert.assertTrue(rf instanceof ReadBase);
+        readBaseFeature = (ReadBase) rf;
+        Assert.assertEquals(1, readBaseFeature.getPosition());
+        Assert.assertEquals(readBases[0], readBaseFeature.getBase());
+        Assert.assertEquals(scores[0], readBaseFeature.getQualityScore());
+    }
+}
```

Comment:
We should add unit tests for check for quality score features as well as the Substitution and ReadBase checks already here. Also, we should have a round-trip test(s) in addition to these unit tests to make sure that we correctly round-trip all these cases.